infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -346,7 +347,18 @@ static void restartServer(boolean resetData) {
     private static void cleanShutdown() {
         log.info("CleanShutdown: Starting Cleanup.");
         shutdownServer = true;
-        activeServer.close();
+        try {
+            CorfuServerNode current = activeServer;
```

Comment:
Why do you need a local variable for this?

Reply:
activeServer is a volatile and can be re-assigned by multiple threads.